src/core/src/ffi/minhash.rs

```diff
@@ -37,6 +37,10 @@ pub unsafe extern "C" fn kmerminhash_new(
         HashFunctions::murmur64_DNA
     };
 
+    let mx: u64 = match max_hash_for_scaled(scaled) {
+        None => 0,
+        Some(max_hash) => max_hash,
+    };
     let mh = KmerMinHash::new(n, k, hash_function, seed, mx, track_abundance);
```

Comment:
since we are changing the APIs, I think we can make `KmerMinHash::new` and `kmerminhash_new` look more like each other. Maybe
```rust
#[no_mangle]
pub unsafe extern "C" fn kmerminhash_new(
    scaled: u64,
    k: u32,
    hash_function: HashFunctions,
    seed: u64,
    track_abundance: bool,
    n: u32,
) -> *mut SourmashKmerMinHash {
```
and
```
impl KmerMinHash {
    pub fn new(
        scaled: u64,
        ksize: u32,
        hash_function: HashFunctions,
        seed: u64,
        track_abundance: bool,
        num: u32,
    ) -> KmerMinHash {
```

This will change how `lib.kmerminhash_new` is called in Python. more like:
```python
hash_function = lib.HASH_FUNCTIONS_MURMUR64_DAYHOFF # or DNA, HP, PROTEIN
self._objptr = lib.kmerminhash_new(
    scaled, ksize, hash_function, seed, track_abundance, n
)
```

Eventually we want to expose `HashFunctions` properly to Python (see #751 for discussion), but for now we can use `lib.HASH_FUNCTIONS_MURMUR64_*` directly.